## Gradient of developmental and injury-response transcriptional states define functional vulnerabilities underpinning glioblastoma heterogeneity

*Laura M. Richards**, *Owen K.W. Whitley**, *Graham MacLeod, Florence M.G. Cavalli, Fiona J. Coutinho , Julia E. Jaramillo, Nataliia Svergun, Mazdak Riverin, Danielle C. Croucher, Michelle Kushida, Kenny Yu, Paul Guilhamon, Naghmeh Rastegar, Moloud Ahmadi, Danielle A. Bozek, Naijin Li, Erika Luis, Nicole I. Park, Julian Spears, Michael D. Cusimano, Benjamin Haibe-Kains, Mathieu Lupien, H. Artee Luchman, Samuel Weiss, Stephane Angers* *Peter B. Dirks**, *Gary D. Bader**, *Trevor J. Pugh**. *Nature Cancer (2020): [insert link to paper once published]()*


Glioblastomas (GBM) harbour diverse populations of cells, including a rare subpopulation of glioblastoma stem cells (GSCs) that drive tumorigenesis. To characterize functional diversity within the tumor-initiating fraction of GBM, we performed single-cell RNA-sequencing on >69,000 GSCs cultured from the tumors of 26 patients. We observed a high degree of inter- and intra-GSC transcriptional heterogeneity that could not be fully explained by somatic alterations at the DNA level. Instead, we found GSCs mapped along a transcriptional gradient spanning two cellular states reminiscent of normal neural developmental and inflammatory wound response processes. Genome-wide CRISPR/Cas9 drop-out screens independently recapitulated this observation, with each state characterized by unique essential genes. Further single-cell RNA-sequencing of >56,000 malignant cells from primary tumors found the majority of cancer cells organize along an astrocyte maturation gradient orthogonal to the GSC gradient yet retained expression of founder GSC transcriptional programs. Our work supports a model whereby GBM grows out of a fundamental GSC-based neural tissue regeneration and wound response transcriptional program characterized by a range of sensitive targets, promising for novel therapy development. 

#
### Where can I find the raw data?

| Data type | Cohort | File type | Location | 
| ----- | ----- |----- |----- |
| Single cell RNA-sequencing | GSCs (XXX cells from 29 samples) | <p>fastq<br>Processed data</p> | <p>EGA<br>Broad Portal</p> | 
| Single cell RNA-sequencing | Tumours (XXXX cells from 7 samples) | <p>fastq<br>Processed data</p> | <p>EGA<br>Broad Portal</p> |
| Single nuclei RNA-sequencing | Tumours (XXX cells from 10 samples) | fastq | 
| Public single cell/nuclei RNA-seq ||
| Bulk RNA-sequencing | GSCs (72 samples) | fastq | | 
| Whole genome sequencing | GSCs + matched normal blood (20 samples) | fastqs | 
| CRISPR screens | GSCs (11 samples) | <p>raw BF<br>qBF</p>| 
| Clinical and experimental metadata | 
  
#
### Who do I contact?
Drs. Trevor Pugh (trevor.pugh@utoronto.ca) & Gary Bader (gary.bader@utoronto.ca)
   



