## Gradient of developmental and injury-response transcriptional states defines roots of glioblastoma heterogeneity

***Laura M. Richards**\*, Owen Whitley\*, Graham MacLeod, Florence M.G. Cavalli, Fiona J. Coutinho , Michelle Kushida, Nataliia Svergun, Danielle Croucher, Kenny Yu, Naghmeh Rastegar, Moloud Ahmadi, Danielle A. Bozek, Julia E. Jaramillo, Naijin Li, Erika Luis, Nicole I. Park, Julian Spears, Michael D. Cusimano, Benjamin Haibe-Kains, H. Artee Luchman, Samuel Weiss, Stephane Angers, Peter B. Dirks#, Gary D. Bader#, **Trevor J. Pugh#*** [insert link to paper once published]()

Glioblastomas (GBM) harbour diverse populations of cells, including a rare subpopulation of glioblastoma stem cells (GSCs) that drive tumorigenesis. To characterize functional diversity within the tumor-initiating fraction of GBM, we performed single-cell RNA-sequencing on >69,000 GSCs cultured from the tumors of 26 patients. We observed a high degree of inter- and intra-GSC transcriptional heterogeneity that could not be fully explained by somatic alterations at the DNA level. Instead, we found GSCs mapped along a  transcriptional gradient spanning two cellular states reminiscent of normal neural developmental and inflammatory wound response processes. Genome-wide CRISPR/Cas9 drop-out screens recapitulated this observation, as essential genes within each GSC comprised a mixture of these states depending on the position along this axis as measured by gene expression profiling. Further single-cell RNA-sequencing of patient tumors found the majority of cancer cells organize along an astrocyte maturation gradient orthogonal to the GSC gradient yet retained expression of founder transcriptional programs found in GSCs. Our work supports a model whereby GBM heterogeneity is rooted in a fundamental GSC-based neural tissue regeneration and wound response transcriptional continuum.

#
### Where can I find the raw data?

| Data type | Cohort | File type | Location |  
| ----- | ----- |----- |----- |
| Single cell RNA-sequencing | GSCs (XXX cells from 29 samples) | fastq | 
| Single cell RNA-sequencing | Tumours (XXXX cells from 7 samples) | fastq | 
| Single nuclei RNA-sequencing | Tumours (XXX cells from 10 samples) | fastq | 
| Bulk RNA-sequencing | GSCs (72 samples) | fastq | | 
| Whole genome sequencing | GSCs + matched normal blood (20 samples) | fastqs | 
| 
  
#
### Who do I contact?
Laura Richards (lauram.richards@mail.utoronto.ca) & Trevor Pugh (trevor.pugh@utoronto.ca)  
   



