---
title: "Feature Selection, RNA"
output:
  html_document:
    df_print: paged
    toc: TRUE
---

## Packages
``` {r}
library(ggplot2)
library(SummarizedExperiment)
library(reticulate)
LM <- reticulate::import('sklearn.linear_model')
```

## functions 
``` {r}
fit_logistic <- function(X, y) {
  ## fit an L1 regularized logistic regression model
  output.model <- LM$LogisticRegressionCV(cv = 5L,
                                          Cs = 50L,
                                         solver = 'liblinear',
                                         penalty = 'l1',
                                         random_state = 32L)

  output.model$fit(X = X, y = y)
  avg.score <- lapply(output.model$scores_, FUN = function(x) {
    apply(x, MARGIN = 2, mean)
  })
  coefs <- output.model$coef_
  coefs <- as.numeric(coefs)
  names(coefs) <- colnames(X)
  
  # check for non-zero coefs. these are selected features
  selected.features <- names(coefs)[which(coefs > 0)]
  selected.features <- selected.features[order(selected.features)]
  
  output_list = list(model = output.model,
                     avg.score = avg.score,
                     selected.features = selected.features)
}

plot_cv_avg <- function(output_list, class.use = 1L, xlab = 'log10(C)', ylab = 'score') {
  # class.use = either integer for indexing avg.score list
  # or a string indicating name of class.
  c.vals <- as.numeric(output_list$model$Cs_)
  best.c <- as.numeric(output_list$model$C_)
  avg.score <- output_list$avg.score[[class.use]]
  df <- data.frame(log10c = log10(c.vals), 
                   avg.score = avg.score, 
                   best.model = c.vals == best.c)
  cols <- c("FALSE" = "blue", "TRUE" = "red")
  p <- ggplot(df, aes(log10c, avg.score, best.model)) + 
    geom_line(aes(x = log10c, y = avg.score), color = 'blue') + 
    geom_point(aes(x = log10c, y = avg.score, color = best.model)) +
    scale_color_manual(values = cols) + 
    xlab(xlab) + ylab(ylab)
  return(p)
}

cor_hist <- function(x, main = 'histogram of correlation values') {
  mat <- cor(x)
  v <- c()
  for (i in 1:(nrow(mat) - 1)) {
    for(j in (i+1):ncol(mat)) {
      v <- c(v, mat[i,j])
    }
  }
  hist(v, main = main)
}

```

## Setup directories
``` {r}
top.dir <- '/home/owenwhitley/projects/su2c_v2/'
scripts.dir <- file.path(top.dir, 'scripts/RNA/analysis')
```

## Fit Models

``` {r}
## RNA-seq differentially expressed genes for GSC clustering analysis

RNA.results.dir <- file.path(top.dir, 'results/RNA')
analysis.results.dir <- file.path(RNA.results.dir, 'RNA_GSCs_clustering_DE')
load(file.path(analysis.results.dir, 'RNA_GSC_obj.RData'))
# use vst gene expression data
gene.expr.GSCs <- t(SummExpDR::assay(RNA.GSC.obj, 'vst'))
# our output is cluster label
meta.data <- as.data.frame(SummExpDR::colData(RNA.GSC.obj))

all(rownames(meta.data) == rownames(gene.expr.GSCs))

RNA.GSC.c1.upreg <- read.csv(file.path(analysis.results.dir, 'spectral_k2_c1_upreg_FDR_0_05.csv'), stringsAsFactors = F)
RNA.GSC.c2.upreg <- read.csv(file.path(analysis.results.dir, 'spectral_k2_c2_upreg_FDR_0_05.csv'), stringsAsFactors = F)
head(RNA.GSC.c1.upreg)
head(RNA.GSC.c2.upreg)
RNA.GSC.c1.upreg <- as.character(RNA.GSC.c1.upreg$x)
RNA.GSC.c2.upreg <- as.character(RNA.GSC.c2.upreg$x)
```

### GSC c1 (Proneural) genes

``` {r}
## L1 regularized model trained using GSC c1 (Proneural) upregulated genes
GSC.c1.model <- fit_logistic(X = gene.expr.GSCs[,RNA.GSC.c1.upreg], y = as.integer(meta.data$spectral_k2))
plot_cv_avg(GSC.c1.model, ylab = 'avg accuracy')
GSC.c1.model$selected.features
hist(GSC.c1.model$model$coef_[RNA.GSC.c1.upreg %in% GSC.c1.model$selected.features],
     main = 'histogram of non-zero coefficients')
cor_hist(gene.expr.GSCs[,GSC.c1.model$selected.features], main = 'correlation of c1 selected features')
```

### GSC c2 upregulated genes

``` {r}
# Same, using GSC c2 upregulated genes
GSC.c2.model <- fit_logistic(X = gene.expr.GSCs[,RNA.GSC.c2.upreg], y = as.integer(meta.data$spectral_k2))
plot_cv_avg(GSC.c2.model, ylab = 'avg accuracy')
GSC.c2.model$selected.features
hist(GSC.c2.model$model$coef_[RNA.GSC.c2.upreg %in% GSC.c2.model$selected.features],
     main = 'histogram of non-zero coefficients')
cor_hist(gene.expr.GSCs[,GSC.c2.model$selected.features], main = 'correlation of c2 selected features')
```

### Tissue vs Line (Tumour vs GSC), GSC upregulated genes

``` {r}
# Same, for GSC upregulated genes, to distinguish Tumour from line
tissue.line.dir <- file.path(RNA.results.dir, 'RNA_tissue_line_clustering')
load(file.path(tissue.line.dir, "RNA_GBM_obj.RData"))
gene.expr.tissue.line <- t(SummExpDR::assay(RNA.GBM.obj, 'vst'))
meta.data.tissue.line <- SummExpDR::colData(RNA.GBM.obj)

GSC.line.upreg <- read.csv(file.path(tissue.line.dir, "line_upreg_FDR_0_05.csv"))
head(GSC.line.upreg)
GSC.line.upreg <- as.character(GSC.line.upreg$x)

tissue.line.model <- fit_logistic(X = gene.expr.tissue.line[,GSC.line.upreg], 
                                  y = as.integer(meta.data.tissue.line$Type_bis))
plot_cv_avg(tissue.line.model, ylab = 'avg accuracy')
tissue.line.model$selected.features
hist(tissue.line.model$model$coef_[GSC.line.upreg %in% tissue.line.model$selected.features],
     main = 'histogram of non-zero coefficients')
cor_hist(gene.expr.tissue.line[,tissue.line.model$selected.features], 
         main = 'correlation of GSC line selected features')
```

## save resulting gene signatures
``` {r}
output.dir <- file.path(RNA.results.dir, 'feature_selection')
if (!dir.exists(output.dir)) {
  dir.create(output.dir)
}

write.csv(GSC.c1.model$selected.features,
          file = file.path(output.dir, 'RNA_GSC_c1_upreg_classifier_genes.csv'))
write.csv(GSC.c2.model$selected.features,
          file = file.path(output.dir, 'RNA_GSC_c2_upreg_classifier_genes.csv'))
write.csv(tissue.line.model$selected.features,
          file = file.path(output.dir, 'RNA_GSC_line_upreg_classifier_genes.csv'))
```

## session Info
``` {r}
sessionInfo()
```
