---
title: "Clustering and Differential Expression, RNA-seq, SU2C adult GBM lines"
output:
  html_document:
    toc: true
    df_print: paged
---

## Setup Directories + Load Data

```{r, results='hide'}
## get required packages
library(DESeq2)
library(SummExpDR)
library(SummarizedExperiment)
## Setup directories
top.dir <- '/home/owenwhitley/projects/su2c_v2/'
scripts.dir <- file.path(top.dir, 'scripts/RNA/analysis')
RNA.results.dir <- file.path(top.dir, 'results/RNA')
analysis.results.dir <- file.path(RNA.results.dir, 'RNA_tissue_line_clustering')

overwrite <- T
if (overwrite) {
  if (dir.exists(analysis.results.dir)) {
    system(paste('rm -rf', analysis.results.dir))
  }
}

for (dir.name in c(RNA.results.dir, analysis.results.dir)) {
  if (!dir.exists(dir.name)) {
    dir.create(dir.name)
  }
}
setwd(scripts.dir)
```

```{r}
data.dir <- file.path(top.dir, 'data/preprocessed/RNA/SU2C_coh1234_nonSU2C/')
## Load data
load(file = file.path(data.dir, 'RNA_adult_GBM_vst_combat_nonSU2C.RData'))
load(file = file.path(data.dir, 'RNA_adult_GBM_dds.RData'))
load(file = file.path(data.dir, 'RNA_adult_GBM_fpkm_combat_nonSU2C.RData'))
dir(data.dir)

# ## run checks
# stopifnot(all(rownames(RNA_adult_GBM_dds) == rownames(RNA_adult_GBM_vst_combat_nonSU2C)))
# stopifnot(all(rownames(RNA_adult_GBM_fpkm_combat_nonSU2C) == rownames(RNA_adult_GBM_vst_combat_nonSU2C)))
# stopifnot(all(colnames(RNA_adult_GBM_dds) == colnames(RNA_adult_GBM_vst_combat_nonSU2C)))
# stopifnot(all(colnames(RNA_adult_GBM_fpkm_combat_nonSU2C) == colnames(RNA_adult_GBM_vst_combat_nonSU2C)))
# stopifnot(identical(colData(RNA_adult_GBM_fpkm_combat_nonSU2C), colData(RNA_adult_GBM_vst_combat_nonSU2C)))
# stopifnot(all(sapply(colnames(colData(RNA_adult_GBM_dds)), FUN = function(x) {
#   if (any(is.na(colData(RNA_adult_GBM_dds)[,x]))) {
#   # skiep NA entries in DESeq2 dataset object
#     y <- T
#   } else {
#     y <- all(colData(RNA_adult_GBM_dds)[,x] == colData(RNA_adult_GBM_vst_combat_nonSU2C)[,x])
#   }
# })))
## Make SummarizedExperiment object
summ.expt <- SummarizedExperiment(assays = list(counts = counts(RNA_adult_GBM_dds),
                                                vst = SummarizedExperiment::assay(RNA_adult_GBM_vst_combat_nonSU2C, 1),
                                                log2.fpkm = log2(SummarizedExperiment::assay(RNA_adult_GBM_fpkm_combat_nonSU2C, 1))),
                                  colData = SummarizedExperiment::colData(RNA_adult_GBM_vst_combat_nonSU2C))
summ.expt

## add nonSU2C status as batch variable

RNA.GBM.obj <- SummExpDR::create_SummExpDR(summ.expt)

```

## Run PCA + clustering

``` {r}
## find + remove outliers
RNA.GBM.obj <- SummExpDR::detect_outliers(RNA.GBM.obj, assay_use = 'vst')

RNA.GBM.obj <- SummExpDR::runPCA(RNA.GBM.obj, i = 'vst', std_norm = TRUE)

SummExpDR::plotDR(RNA.GBM.obj, 'PC1', 'PC2', 'PCA', color_by = 'Cohort', shape_by = 'Lab')
SummExpDR::plotDR(RNA.GBM.obj, 'PC1', 'PC2', 'PCA', color_by = 'is.outlier', shape_by = 'Lab', label = 'SampleID_used', lab_size = 2.0)

SummExpDR::plotDR(RNA.GBM.obj, 'PC2', 'PC3', 'PCA', color_by = 'is.outlier', shape_by = 'Lab', label = 'SampleID_used', lab_size = 2.0)

SummExpDR::plotDR(RNA.GBM.obj, 'PC2', 'PC4', 'PCA', color_by = 'is.outlier', shape_by = 'Lab', label = 'SampleID_used', lab_size = 2.0)

SummExpDR::screePlot(RNA.GBM.obj, key = 'PCA', dims_use = paste0('PC', 1:20))

col_data_w_outliers <- SummExpDR::colData(RNA.GBM.obj)
outlier_samples <- rownames(col_data_w_outliers)[col_data_w_outliers$is.outlier]
# BT100_T is missed as an outlier
outlier_samples <- c(outlier_samples, col_data_w_outliers$Sample[col_data_w_outliers$SampleID_used == 'BT100_T'])

length(outlier_samples)
outlier_samples

RNA.GBM.obj <- SummExpDR::subsetData(RNA.GBM.obj, cols = setdiff(rownames(col_data_w_outliers), outlier_samples))
RNA.GBM.obj <- SummExpDR::runPCA(RNA.GBM.obj, i = 'vst', std_norm = TRUE)

SummExpDR::plotDR(RNA.GBM.obj, 'PC1', 'PC2', 'PCA', color_by = 'Type_bis', shape_by = 'Lab')
```

``` {r}
RNA.GBM.obj <- SummExpDR::run_multi_k_functions(x = RNA.GBM.obj, analysis_name = 'spectral', 
                                               k_use = 2:4, num_iter = 200, thresh_ARI = 0.80,
                                               thresh_proportion = 0.80, assay_use = 'vst',
                                               algorithm = 'spectral_kernlab', std_norm = T,
                                               sigma = 0.5, affi_K = 10)

SummExpDR::plot_sim_dist(RNA.GBM.obj, save_file = F, mode = 'full.vs.rs', clust_analysis = 'spectral')
SummExpDR::plot_sim_dist(RNA.GBM.obj, save_file = F, mode = 'resample.vs.perm', clust_analysis = 'spectral')

RNA.GBM.obj <- SummExpDR::get_soln_k(RNA.GBM.obj, add_2_metadata = T, k = 2, pick_best_valid = F)

SummExpDR::plotDR(RNA.GBM.obj, 'PC1', 'PC2', 'PCA', color_by = 'spectral_k2', shape_by = 'Lab')

```

## Compare clustering results to those obtained on GSCs only, assess correlation of results.

``` {r}
## show clustering rsults from GSC lines in context of Tumor + line
# directory containing results for RNA GSCs clustering + Diff Exp analysis
RNA.GSCs.results.dir <- file.path(analysis.results.dir, '../RNA_GSCs_clustering_DE')
load(file = file.path(RNA.GSCs.results.dir, 'RNA_GSC_obj.RData'))
GSC.metadata <- SummExpDR::colData(RNA.GBM.obj)
GSC.clust.k2.spectral <- GSC.metadata$spectral_k2
names(GSC.clust.k2.spectral) <- rownames(GSC.metadata)
RNA.GBM.obj <- SummExpDR::addColData(RNA.GBM.obj, GSC.clust.k2.spectral, col_name = 'GSC.clust')

```

## Differential Expression + GSEA

#### Run Differential Expression between tissue + line
``` {r}

# run differential expression
RNA.GBM.obj <- SummExpDR::run_diff_exp(expt_data = RNA.GBM.obj, class_type_use = 'Type_bis', covariates = c('is.nonSU2C'),
                                        class1 = 'T', class2 = 'L', pipeline = 'DESEQ2', write_output = T,
                                        output_dir = analysis.results.dir, prefix = 'RNA_GBM_',
                                        n_cores = 1)
# get marker genes and compare results to previous results
# shown at SU2C October 3 Face to Face meeting
diff.exp.res <- SummExpDR::getAnalyses(RNA.GBM.obj, key = 'RNA_GBM_Type_bis_T_v_L')$diff_exp
is.upreg <- diff.exp.res$log2FoldChange > 0
is.downreg <- diff.exp.res$log2FoldChange < 0
passes.fdr <- diff.exp.res$padj < 0.05
tissue.markers <- rownames(diff.exp.res[is.upreg & passes.fdr, ])
line.markers <- rownames(diff.exp.res[is.downreg & passes.fdr, ])

# save 'marker genesets' in results directory
write.csv(tissue.markers, file = file.path(analysis.results.dir, 'tissue_upreg_FDR_0_05.csv'), quote = F, row.names = F)
write.csv(line.markers, file = file.path(analysis.results.dir, 'line_upreg_FDR_0_05.csv'), quote = F, row.names = F)

```


#### Run GSEA
``` {r}
dir(analysis.results.dir)

files <- dir(file.path(analysis.results.dir, 'RNA_GBM_Type_bis_T_v_L'))
rankfile <- files[grep('.rnk', files)]

SummExpDR::enrich_GSEA(gsea_rank_list_path = file.path(analysis.results.dir, 'RNA_GBM_Type_bis_T_v_L',
                                            rankfile),
                        gsea_analysis_name = 'RNA_GBM_Type_bis_T_v_L',
                        gsea_out = analysis.results.dir,
                        gsea_gmx = '../../../data/raw/gmt_files/baderlab_gmts/Human_GO_AllPathways_no_GO_iea_April_01_2018_symbol.gmt')

```

#### Examine results 

``` {r}
dir.contents <- dir(analysis.results.dir)
GSEA.path <- file.path(analysis.results.dir, dir.contents[grep('GseaPreranked', dir.contents, ignore.case = T)])
if (length(GSEA.path) != 1) {
  stop(paste('expected 1 match for GSEA.path'))
}

GSEA.results <- retrieve_gsea(GSEA.path)
```

__Examine Enriched Pathways__
``` {r}
## Positively Enriched
GSEA.results$na_pos_report
## Negatively Enriched
GSEA.results$na_neg_report
```

#### Save Results
__Save Clustering + DiffExp Results to File__
``` {r}
save(RNA.GBM.obj, file = file.path(analysis.results.dir, 'RNA_GBM_obj.RData'))
```

## Session Info

``` {r}
Sys.time()
sessionInfo()
```
