---
title: "Make custom gmt file"
output: html_notebook
---


```{r}
###############################################
########### make GMT file #####################
require(biomaRt)
require(org.Hs.eg.db)

## setup output directory
output_dir <- '~/GSEA/custom_gmts'
if (!dir.exists(output_dir))
{
  dir.create(output_dir)
}

### setup all mappings of ids, terms, and symbols
head(org.Hs.egGO2ALLEGS)
## mappings of entrez ids to GO IDs
GO_mappings <- toTable(org.Hs.egGO2ALLEGS)
GO_mappings <- GO_mappings[order(GO_mappings$go_id),]
print('GO Mappings')
head(GO_mappings)
# gene_id      go_id Evidence Ontology
# 20563      142 GO:0000002      IMP       BP
# 43479      291 GO:0000002      TAS       BP
# 238806    1763 GO:0000002      IDA       BP
# 256261    1890 GO:0000002      IMP       BP
# 521698    3980 GO:0000002      IMP       BP
# 548638    4205 GO:0000002      ISS       BP

## go ids
go_ids <- unique(GO_mappings$go_id)
head(go_ids)
# [1] "GO:0000002" "GO:0000003" "GO:0000009" "GO:0000010" "GO:0000011" "GO:0000012"
## mapping of go ids to go terms
term_mappings <- AnnotationDbi::select(GO.db::GO.db, 
                                       keys = go_ids, 
                                       columns = "TERM")
print('term mappings')
head(term_mappings)
# GOID                                      TERM
# 1 GO:0000002          mitochondrial genome maintenance
# 2 GO:0000003                              reproduction
# 3 GO:0000009    alpha-1,6-mannosyltransferase activity
# 4 GO:0000010 trans-hexaprenyltranstransferase activity
# 5 GO:0000011                       vacuole inheritance
# 6 GO:0000012                single strand break repair
## entrez_ids
entrez_ids <- unique(GO_mappings$gene_id)
# any(duplicated(entrez_ids))
# [1] FALSE
## mapping of go ids to ontology
GO_id_ontology_mapping <- GO_mappings[,c('go_id','Ontology')]
GO_id_ontology_mapping <- GO_id_ontology_mapping[which(!duplicated(GO_id_ontology_mapping$go_id)),]
rownames(GO_id_ontology_mapping) <- GO_id_ontology_mapping$go_id
print('GO ontology mapping')
head(GO_id_ontology_mapping)
# go_id Ontology
# GO:0000002 GO:0000002       BP
# GO:0000003 GO:0000003       BP
# GO:0000009 GO:0000009       MF
# GO:0000010 GO:0000010       MF
# GO:0000011 GO:0000011       BP
# GO:0000012 GO:0000012       BP


## mapping of entrez ids to hgnc symbols
fname <- file.path(output_dir, 'entrez_hgnc_mappings.RData')
if (!file.exists(fname)) {
  ensembl=useMart("ensembl")
  ensembl_HS = useDataset("hsapiens_gene_ensembl",mart=ensembl)
  attributes = listAttributes(ensembl_HS)
  # attributes[grep('entrez', attributes$name),]
  # name                   description         page
  # 56 entrezgene_trans_name EntrezGene transcript name ID feature_page
  # 77            entrezgene                  NCBI gene ID feature_page
  # attributes[grep('hgnc', attributes$name),]
  # name             description         page
  # 61         hgnc_id                 HGNC ID feature_page
  # 62     hgnc_symbol             HGNC symbol feature_page
  # 63 hgnc_trans_name HGNC transcript name ID feature_page
  get_BM_out <- getBM(attributes = c('entrezgene', 'hgnc_symbol'),
                      filters = 'entrezgene',
                      values = entrez_ids,
                      mart = ensembl_HS)
  save(get_BM_out, file = fname)
} else {
  load(fname)
}
entrez_hgnc_mapping <- get_BM_out
any(duplicated((entrez_hgnc_mapping$entrezgene)))
# [1] TRUE
any(duplicated(entrez_hgnc_mapping$hgnc_symbol))
# [1] TRUE

## it appears there are multiple entrezgenes for some hgnc symbols
## and multiple hgnc symbols for some entrez gene ids


## write GMT file. At time of writing, last update to
## org.HS.eg.db and GO.db were March 28, 2018

# fname <- 'org_HS_eg_db_GO_db_Mar_28_2018_v2_min_5_max_200.gmt'
# fname <- 'org_HS_eg_db_GO_db_Mar_28_2018_v2.gmt'



## Run settings
# filter_geneset_size <- T
filter_geneset_size <- F
# ID_type <- 'entrez_id'
ID_type <- 'hgnc_symbol'
pb <- txtProgressBar()

print('Run Settings')
print(paste('ID type used in output:', ID_type))
print(paste('filter geneset size:', filter_geneset_size))
fname <- paste0('org_HS_eg_db_GO_db_Jan_31_19', '_IDType_', ID_type)
if (filter_geneset_size)
{
  
  max_gs_size <- 200
  min_gs_size <- 15
  num_skipped <- 0
  excluded_too_large <- 0
  excluded_too_small <- 0
  print(paste('Max geneset size:', max_gs_size))
  print(paste('Min geneset size:', min_gs_size))
  fname <- paste0(fname, '_maxGSSize_', max_gs_size, '_minGSSize_', min_gs_size)
}
fname <- paste0(fname, '.gmt')
fileConn<-file(file.path(output_dir, fname))
char_vect <- character(0)
print(paste('Output file name:', file.path(output_dir, fname)))

print('Preparing to write file')
for (i in 1:nrow(term_mappings))
{
  curr_id <- term_mappings$GOID[i]
  curr_term <- term_mappings$TERM[i]
  go_ontology <- GO_id_ontology_mapping[curr_id,]
  entrez_ids <- GO_mappings[which(GO_mappings$go_id == curr_id), 'gene_id']
  entrez_ids <- unique(entrez_ids)
  
  if (filter_geneset_size)
  {
    # decide whether to skip geneset. Note that this filters geneset on number
    # of associated entrez ids, not number of associated hgnc symbols
    gs_size <- length(entrez_ids)
    if (gs_size > max_gs_size)
    {
      num_skipped <- num_skipped + 1
      excluded_too_large <- excluded_too_large + 1
      next
    } else if (gs_size < min_gs_size)
    {
      num_skipped <- num_skipped + 1
      excluded_too_small <- excluded_too_small + 1
      next
    }
  }
  if (ID_type == 'hgnc_symbol')
  {
    # subset for hgnc symbols
    matching_entrez_ind <- which(entrez_hgnc_mapping$entrezgene %in% entrez_ids)
    hgnc_symbols_use <- entrez_hgnc_mapping$hgnc_symbol[matching_entrez_ind]
    hgnc_symbols_use <- unique(hgnc_symbols_use)
    gene_ids_use <- hgnc_symbols_use
  } else if(ID_type == 'entrez_id')
  {
    gene_ids_use <- entrez_ids
  } else {
    stop(paste('ID_type', ID_type, 'not recognized. Must be hgnc_symbol or entrez_id'))
  }
  
  
  gene_id_string <- paste(gene_ids_use, collapse = '\t')
  # add new line to character vector
  new_line <- paste(curr_id, curr_term, gene_id_string, sep = '\t')
  char_vect <- c(char_vect, new_line)
  # update progress bar
  setTxtProgressBar(pb, value = (i/nrow(term_mappings)))
}
if (exists('num_skipped'))
{
  print('number of genesets excluded:')
  print(num_skipped)
  print('number of genesets above upper threshold:')
  print(excluded_too_large)
  print('number of genesets below lower threshold:')
  print(excluded_too_small)
}
print('Writing File')
writeLines(text = char_vect, con = fileConn, sep = '\n')
print('File Written')

sessionInfo()
```

