---
title: "Get Preprocessed Adult GBM Samples"
output: html_notebook
---

``` {r}
#####################################################################################################
############### Make Analysis Ready RNA-seq Data for Adult GBM Samples, coh1234 + nonSU2C ###########
#####################################################################################################

## OVERVIEW:  Make DESEQ2 dataset objects as well as vst transformed data for
##            adult GBM samples. Note that there were large batch effects observed
##            previously with th nonSU2C samples, so I elect to perform batch correction
##            for nonSU2C status. note that for cohort 4 we only hav e1 sample, and 
##            from prior analyses, we see that cohort 4 is not strongly separated from
##            cohorts 1-3 in the space of RNA-seq data. The batch effects for nonSU2C
##            status were likely due to differences in sequencing depth as well as
##            individual who performed experiments, according to collaborators
##            involved in SU2C project (e.g. Florence Cavalli at Taylor Lab)
##            Note that here, we only use GBM tumour samples.
##
##          files used include: 
##              RNA_coh1234_nonSU2C_counts_SummExp.RData:
##              counts and metadata for RNA cohorts 1-4 + nonSU2C samples
##              exonic.gene.sizes_chr.RData:
##              exonic gene sizes for hg38, generated by Florence Cavalli
##          see associated README.txt files for respective directories if you want 
##          further information regarding data

#############################################################################################
print('BEGIN SCRIPT')
print(Sys.time())
### Load required packages + functions
library(readxl)
library(SummarizedExperiment)
library(DESeq2)
library(sva)
### Load in Data
# data_folder <- '../data_01_2018'
## setup directories
top_dir <- '~/projects/su2c_v2'
# ## metadata directory
# preproces_metadata_dir <- file.path(top_dir, 'data/preprocessed/metadata')
## directory to output preprocessed RNA data, and location of SummarizedExperimentObject
preprocess_RNA_dir <- file.path(top_dir, 'data/preprocessed/RNA')
stopifnot(dir.exists(preprocess_RNA_dir))
preprocess_RNA_coh1234 <- file.path(preprocess_RNA_dir, 'SU2C_coh1234_nonSU2C')
stopifnot(dir.exists(preprocess_RNA_coh1234))
## Load RNA SummarizedExperiment object
fname_RNA_SummExp <- 'RNA_coh1234_nonSU2C_counts_SummExp.RData'
load(file = file.path(preprocess_RNA_coh1234, fname_RNA_SummExp))
RNA_SummExp <- RNA_coh1234_nonSU2C_counts_SummExp
rm(RNA_coh1234_nonSU2C_counts_SummExp)
###############################################################################
### Subset for adult GBM samples that aren't technical replicates
## recall that SampleID used corresponds to biological sample,
## Sample refers to a particular experiment performed. Not the best nomenclature,
## but changing metadata labels now may make things more confusing
## order by cohort 
RNA_SummExp <- RNA_SummExp[, order(RNA_SummExp$Cohort,
                                   RNA_SummExp$Type_bis,
                                   RNA_SummExp$SampleID_used)]
# unique(RNA_SummExp$Cohort)
# [1] "Cohort1" "Cohort2" "Cohort3" "Cohort4" "nonSU2C"
##
is.adult <- RNA_SummExp$Age_group == 'adult'
# unique(RNA_SummExp$Age_group)
# "adult"     "pediatric" "fetal"   
is.gbm <- RNA_SummExp$Tumor_type == 'GBM'
is.line <- RNA_SummExp$Type_bis == 'L'
# unique(RNA_SummExp$Tumor_type)
# [1] "GBM"     "PFA"     "control"
is.tech.rep <- duplicated(RNA_SummExp$SampleID_used)
is.nonSU2C <- RNA_SummExp$Cohort == 'nonSU2C'
# remove G607 and G811 as 607 is improperly annotated as a GBM,
# G811 may be improperly annotated as a GBM. These samples 
# were not included in analysis as of Oct 2018 committee meeting
G811.or.G607 <- grepl('G607|G811', RNA_SummExp$SampleID_used)

## check that any duplicates in cohorts 1-4 are not in nonSU2C set
## (i.e. if there are duplicates between SU2C and nonSU2C set, 
## nonSU2C labeled as duplicate and will be removed)
SU2C_dups <- RNA_SummExp$SampleID_used[which(is.tech.rep & !is.nonSU2C)]
if(any(SU2C_dups %in% RNA_SummExp$SampleID_used[which(is.nonSU2C)])) {
  stop('expectation failed, expect that all samples listed as duplicate in
       cohorts 1-4 are not listed as duplicates of nonSU2C samples. If
       there are duplicate samples between coh1-4 and nonSU2C cohort,
       nonSU2C sample should be marked as duplicated')
}
## perform subsetting
RNA_adult_GBM_tumour <- RNA_SummExp[,which(is.adult & is.gbm & !is.line & !is.tech.rep & !G811.or.G607)]

#########################################################################
### Subset genes
## load in gene lengths
raw_RNA_dir <- file.path(top_dir, 'data/raw/RNA')
raw_coh123 <- file.path(raw_RNA_dir, 'RNA_florence_01_2018_SU2C_coh123')
load(file.path(raw_coh123, 'exonic.gene.sizes_chr.RData'))

exonic.gene.sizes <- as.numeric(exonic.gene.sizes_chr)
equiv.entries <- sapply(1:length(exonic.gene.sizes_chr), FUN = function(x) {
  exonic.gene.sizes[x] == exonic.gene.sizes_chr[[x]]
})
if(!all(equiv.entries)) {
  stop('expected the coercing exonic.gene.sizes_chr to numeric would preserve
       ordering, assumption failed')
}
names(exonic.gene.sizes) <- names(exonic.gene.sizes_chr)

common.genes <- intersect(rownames(RNA_adult_GBM_tumour), names(exonic.gene.sizes))
RNA_adult_GBM_tumour <- RNA_adult_GBM_tumour[rownames(RNA_adult_GBM_tumour) %in% common.genes,]
exonic.gene.sizes <- exonic.gene.sizes[rownames(RNA_adult_GBM_tumour)]
rowData(RNA_adult_GBM_tumour)$basepairs <- exonic.gene.sizes
## remove genes with less than 5 counts across all samples
RNA_adult_GBM_tumour <- RNA_adult_GBM_tumour[which(apply(assay(RNA_adult_GBM_tumour), 
                                           MARGIN = 1, 
                                           FUN = sum) >= 5),]

#########################################################################
### Make DESEQ2 object and get fpkm values and vst transform
print('==============================================')
print('making DESEQ2 object for non-SU2C data, getting fpkm values, vst transform')


## make DESEQ2 dataset object
RNA_adult_GBM_tumour_dds <- DESeqDataSet(se = RNA_adult_GBM_tumour,
                                       design = ~1)

RNA_adult_GBM_tumour_dds <- estimateSizeFactors(RNA_adult_GBM_tumour_dds)
fname <- 'RNA_adult_GBM_tumour_dds.RData'
save(RNA_adult_GBM_tumour_dds, file = file.path(preprocess_RNA_coh1234, fname))
## set all categorical values to be used in modeling to factors
colData(RNA_adult_GBM_tumour_dds)$Type_bis <- factor(colData(RNA_adult_GBM_tumour_dds)$Type_bis)
colData(RNA_adult_GBM_tumour_dds)$Lab <- factor(colData(RNA_adult_GBM_tumour_dds)$Lab)
colData(RNA_adult_GBM_tumour_dds)$Sex <- factor(colData(RNA_adult_GBM_tumour_dds)$Sex)
colData(RNA_adult_GBM_tumour_dds)$Primary_recurrent <- factor(colData(RNA_adult_GBM_tumour_dds)$Primary_recurrent)

## get fpkm values, save as SummarizedExperiment objects

RNA_adult_GBM_tumour_fpkm <- DESeq2::fpkm(object = RNA_adult_GBM_tumour_dds, 
                                          robust = T)
RNA_adult_GBM_tumour_fpkm <- SummarizedExperiment(assays = list(fpkm = RNA_adult_GBM_tumour_fpkm),
                                                  colData= colData(RNA_adult_GBM_tumour_dds))
fname2 <- 'RNA_adult_GBM_tumour_fpkm.RData'
save(RNA_adult_GBM_tumour_fpkm, file = file.path(preprocess_RNA_coh1234, fname2))

## get vst transform
RNA_adult_GBM_tumour_vst <- vst(RNA_adult_GBM_tumour_dds)
fname3 <- 'RNA_adult_GBM_tumour_vst.RData'
save(RNA_adult_GBM_tumour_vst, file = file.path(preprocess_RNA_coh1234, fname3))

print('END SCRIPT')
print(Sys.time())
print(sessionInfo())
```