---
title: "Preprocess GeneSets"
output:
  html_document:
    df_print: paged
---

``` {r}
###############################################################################
### Compile a database of Cell Type 'Marker Genes' ###

### OVERVIEW: preprocess all genesets of interest we want to use for RNA signature scoring

library(readxl)
library(biomaRt)
```

``` {r}
# Function for filtering genesets
filter_genesets <- function(gene_set_list, 
                            master_geneset = NULL, 
                            min_geneset_size = 1,
                            max_geneset_size = Inf) {
  if (is.null(master_geneset)) {
    print('master_geneset set to NULL, keeping all genes in genesets')
    gene_set_list <- gene_set_list
    original_num_genes <- lapply(gene_set_list, FUN = function(x) length(x))
    genes_kept <- lapply(gene_set_list, FUN = function(x) length(x))
  } else {
    if (!is.character(master_geneset)) {
      stop('must specify geneset for which all markersets must be subsets of')
    }
    # calculate original number of genes, filter for genes in master geneset,
    # and determine number of genes kept for each geneset
    original_num_genes <- lapply(gene_set_list, FUN = function(x) length(x))
    gene_set_list <- lapply(gene_set_list, FUN = function(x) return(x[x %in% master_geneset]))
    genes_kept <- lapply(gene_set_list, FUN = function(x) length(x))
  }
  original_num_genes <- as.numeric(original_num_genes)
  names(original_num_genes) <- names(gene_set_list)
  genes_kept <- as.numeric(genes_kept)
  names(genes_kept) <- names(gene_set_list)
  # make dataframe containing geneset info
  geneset_info <- data.frame(geneset_name = names(gene_set_list),
                             original_num_genes = original_num_genes,
                             genes_kept = genes_kept,
                             stringsAsFactors = FALSE)
  rownames(geneset_info) <- geneset_info$geneset_name
  # remove genesets with fewer genes than min geneset size
  genesets_discard <- (genes_kept < min_geneset_size) | (genes_kept > max_geneset_size)
  if (any(genesets_discard)) {
    print(paste('removing contents for', length(which(genesets_discard)), 'genesets outside acceptable size range'))
    gene_set_list <- gene_set_list[-which(genesets_discard)]  
  }
  return(list(gene_set_list = gene_set_list, 
              geneset_info= geneset_info))
}
```

``` {r}
###############################################################################
### setup directories
setwd('~/projects/su2c_v2/scripts/RNA/preprocessing')
top.dir <- '~/projects/su2c_v2/'
raw.data.folder <- file.path(top.dir, 'data/raw/GeneSets')
preproc.data.folder <- file.path(top.dir, 'data/preprocessed')
output.dir <- file.path(preproc.data.folder, 'GeneSets')
if (!dir.exists(output.dir)) {
  dir.create(output.dir)
}

###############################################################################
### create functions for constructing and filtering genesets ###

## define helper function to check tibble input
## expect dataframe with entreies akin to those found in output of 
## Seurat's FindAllMarkers function
check_tibble <- function(inp_tibble, reqd_columns)
{
  if (!(is(inp_tibble, 'tibble') | is(inp_tibble, 'data.frame')))
  {
    stop('must be tibble or dataframe input')
  } else if (!all(reqd_columns %in% colnames(inp_tibble)) )
  {
    stop(paste('must have', reqd_columns), 'columns')
  }
}

## define function to extract tibble from tibble taken from xls
extract_tibble <- function(inp_tibble)
{
  reqd_columns <- c('avg_diff', 'cluster', 'gene')
  check_tibble(inp_tibble, reqd_columns)
  return(inp_tibble[,reqd_columns])
}


## define function to extract markers for each cluster
## expect dataframe with entries akin to those found in output of 
## Seurat's FindAllMarkers function
extract_cluster_markers <- function(inp_tibble, # tibble or dataframe with average difference, 
                                                # cluster, and gene columns
                                    subtype_name # name to add to cluster names
)
{
  reqd_columns <- c('avg_diff', 'cluster', 'gene')
  check_tibble(inp_tibble, reqd_columns)
  clusters <- unique(inp_tibble$cluster)
  inp_tibble$avg_diff <- as.numeric(inp_tibble$avg_diff)
  marker_set_list <- list()
  cluster_names <- paste0(subtype_name, '_', clusters)
  cluster_names <- sub(' ', '_', cluster_names)
  upreg_names <- paste0(cluster_names,'_upreg')
  downreg_names <- paste0(cluster_names, '_downreg')
  
  ## indices for any genes upregulated in any cluster,
  ## any genes downregulated in any cluster, resplectively
  upreg_inds <- which(inp_tibble$avg_diff > 0)
  downreg_inds <- which(inp_tibble$avg_diff < 0)
  
  for (i in 1:length(clusters))
  {
    ## give names to up and downregulated genesets
    upreg_name_i <- upreg_names[i]
    downreg_name_i <- downreg_names[i]
    ## get indices for given cluster, and then the upreg
    ## and downreg genes for the cluster for genes within our master geneset
    ## also count number of genes in original upregulated set and genes in
    ## filtered geneset
    cluster_i <- clusters[i]
    # indices for markers matching cluster
    cluster_i_inds <- which(inp_tibble$cluster == cluster_i)
    # upregulated gene indices
    cluster_i_upreg_inds <- intersect(upreg_inds, cluster_i_inds)
    # upregulated genes used for further analysis
    cluster_i_upreg <- inp_tibble$gene[cluster_i_upreg_inds]
    # indices for downregulated genes
    cluster_i_downreg_inds <- intersect(downreg_inds, cluster_i_inds)
    # downregulated genes used for further analysis
    cluster_i_downreg <- inp_tibble$gene[cluster_i_downreg_inds]
    
    ## assign genesets to our marker set list
    marker_set_list[[upreg_name_i]] <- cluster_i_upreg
    marker_set_list[[downreg_name_i]] <- cluster_i_downreg
  }
  return(marker_set_list)
}

## function to filter genesets for genes within a 'master geneset'. In our case, the master geneset
## will be all the genes in the fpkm matrix used for scoring.

###############################################################################
### Get set of genes in RNA data used for diff exp analysis, set up global
### variables for markerset list

## prior version used ensembl version 95 hgnc symbols
# human_mart <- useMart("ensembl", dataset = "hsapiens_gene_ensembl", host = "www.ensembl.org",
#                       ensemblRedirect = F)
# # listMarts()
# # biomart               version
# # 1 ENSEMBL_MART_ENSEMBL      Ensembl Genes 95
# # 2   ENSEMBL_MART_MOUSE      Mouse strains 95
# # 3     ENSEMBL_MART_SNP  Ensembl Variation 95
# # 4 ENSEMBL_MART_FUNCGEN Ensembl Regulation 95
# human_BM <- getBM(attributes = c( 'ensembl_gene_id', 'hgnc_symbol'), mart = human_mart)
# hgnc.symbols <- unique(human_BM$hgnc_symbol)
# rm(human_mart, human_BM)

## Load fpkm data object
load(file = file.path(raw.data.folder, '../../preprocessed/RNA/SU2C_coh1234_nonSU2C/RNA_adult_GBM_fpkm.RData'))
SU2C.all.RNA.genes <- rownames(RNA_adult_GBM_fpkm)
save(SU2C.all.RNA.genes, file = file.path(output.dir, 'SU2C_all_RNA_genes.RData'))

all.markers <- list()
```

```{r}
#########################################################################################
### Load in Zhong Markers ###
## Zhong, 2018, A single-cell RNA-seq survey of the developmental landscape of
## the human prefrontal cortex
Zhong_markers_directory <- file.path(raw.data.folder, 'zhong_2018')
fname_xls <- 'nature25980-s2.xlsx'

## major cell types, as difined in paper (derived from clustering using Seurat package)
fpath_xls <- file.path(Zhong_markers_directory, fname_xls)
zhong.major.cell.types <- read_xlsx(path = fpath_xls, sheet = 'Major cell types',
                              skip = 4)
zhong.major.cell.types <- extract_tibble(zhong.major.cell.types)
# head(zhong.major.cell.types)
# # A tibble: 6 x 3
# avg_diff cluster gene 
# <dbl> <chr>   <chr>
#   1     1.60 NPCs    HMGN2
# 2     2.62 NPCs    HMGB2
# 3    -1.60 NPCs    STMN2
# 4     1.92 NPCs    PAX6 
# 5     1.29 NPCs    H2AFZ
# 6     2.11 NPCs    SMC4 
zhong.2018.marker.sets <- extract_cluster_markers(zhong.major.cell.types, 
                                                  'Zhong')

# ## all sheets in the excel file
# sheets_list <- excel_sheets(fpath_xls)
# [1] "Major cell types"         "NPCs"                     "Excitatory neurons"       "Excitatory neurons_WEEK" 
# [5] "Excitatory neurons_stage" "Interneurons"             "OPCs"                     "Astrocytes"              
# [9] "Microglia" 

all.markers <- c(all.markers, zhong.2018.marker.sets)
```

``` {r}
###############################################################################
## Load in Nowakowski marker sets ##
# Nowakowski et al. 2017
nowakowski_marker_genes <- read_xlsx(path = file.path(raw.data.folder,
                                                     'nowakowski_2017',
                                                     'aap8809_Nowakowski_SM-Tables-S1-S11.xlsx'),
                                    sheet = 'Table5 - Clustermarkers')
head(nowakowski_marker_genes)
nowakowski_marker_genes <- extract_tibble(nowakowski_marker_genes)
head(nowakowski_marker_genes)

Nowakowski_marker_sets <- extract_cluster_markers(nowakowski_marker_genes, 
                                                 'nowakowski')

all.markers <- c(all.markers, Nowakowski_marker_sets)
```

``` {r}
################################################################################
### TCGA markersets ###
## Highly expressed genes for each subtype identified in
## Verhaak et al. 2010 An integrated genomic analysis identifies clinically 
## relevant subtypes of glioblastoma characterized by abnormalities in PDGFRA, 
## IDH1, EGFR and NF1
TCGA_folder <- file.path(raw.data.folder, 'verhaak_TCGA_2010')
TCGA_types_file <- 'TCGA_unified_CORE_ClaNC840.txt'
# TCGA_840_genelist_file <- 'ClaNC840_centroids.csv'
TCGA_table <- read.delim(file.path(TCGA_folder, TCGA_types_file), header = F, stringsAsFactors = F)
col_names <- c('gene', 'subtype')
gene_subtype_mapping <- TCGA_table[3:nrow(TCGA_table),1:2]
colnames(gene_subtype_mapping) <- col_names
gene_subtype_mapping$subtype[gene_subtype_mapping$subtype == ''] <- 'None'
# head(gene_subtype_mapping)
# gene subtype
# 3  CDKN1B      PN
# 4   EPB41      PN
# 5    CLGN      PN
# 6  PDE10A      PN
# 7 RALGPS2      PN
# 8    TAF5      PN
TCGA_marker_sets <- list()
TCGA_marker_sets[['TCGA_PN']] <- gene_subtype_mapping$gene[gene_subtype_mapping$subtype == 'PN']
TCGA_marker_sets[['TCGA_NL']] <- gene_subtype_mapping$gene[gene_subtype_mapping$subtype == 'NL']
TCGA_marker_sets[['TCGA_CL']] <- gene_subtype_mapping$gene[gene_subtype_mapping$subtype == 'CL']
TCGA_marker_sets[['TCGA_MES']] <- gene_subtype_mapping$gene[gene_subtype_mapping$subtype == 'MES']

all.markers <- c(all.markers, TCGA_marker_sets)
```

```{r}
###############################################################################
### Cahoy marker sets from mouse to human HGNC
### markersets were derived from supplementary tables
### from Cahoy et al. 2008, A Transcriptome Database for Astrocytes, Neurons, 
### and Oligodendrocytes: A New Resource for Understanding Brain Development 
### and Function. Tables include S4-S6, S15-S21
## Load in Cahoy Datasets

cahoy.data.folder <- file.path(raw.data.folder, 'cahoy_2008')
files <- dir(cahoy.data.folder)
xls_files <- files[grep('xls$', files)]

# xls_files
# [1] "340_Cahoy_S_Table_S4_AvX_2007-09-11.xls"           "350_Cahoy_S_Table_S5_OvX_2007-09-11.xls"          
# [3] "360_Cahoy_S_Table_S6_NvX_2007-09-11.xls"           "450_Cahoy_S_Table_S15_AYvAO_down_2007-09-11.xls"  
# [5] "460_Cahoy_S_Table_S16_AYvAO_up_2007-09-11.xls"     "470_Cahoy_S_Table_S17_OPCvMOG_down_2007-09-11.xls"
# [7] "480_Cahoy_S_Table_S18_OPCvMOG_up_2007-11-30.xls"   "490_Cahoy_S_Table_S19_GCvMOG_up_2007-09-11.xls"   
# [9] "500_Cahoy_S_Table_S20_AvCult_up_2007-09-11.xls"    "510_Cahoy_S_Table_S21_AvCult_down_2007-09-11.xls" 

## manually assign names appropriate to the description of the supplementary tables provided
## in SI for Cahoy et al 2008

## comparisons done in paper were astrocyte vs all others (oligo + neuron), 
## oligo vs all others (astrocyte + neuron), neuron vs all others (astrocyte + oligo),
## OPCs vs mature (MOG+) oligodendrocytes, MOG+ vs MOG- OLs,
## in vivo astrocytes vs cultured astroglia
## all pathways here can be assumed to be upregulated
names_use <- c('cahoy_astrocyte', 'cahoy_oligodendrocyte',
               'cahoy_neuron', 'cahoy_astro_young',
               'cahoy_astro_mature', 'cahoy_OPC',
               'cahoy_OL_myel', 'cahoy_MOG_pos',
               'cahoy_astro_in_vivo', 'cahoy_cultured_astroglia')

genelist <- read_xls(file.path(cahoy.data.folder, xls_files[1]), skip = 1)
# # A tibble: 2,618 x 4
# X__1 `Probe Set ID` `Gene Name` `Fold Enriched In Astrocytes`
# <dbl> <chr>          <chr>                               <dbl>
#   1     1 1440142_s_at   Gfap                                 84.9
# 2     2 1434449_at     Aqp4                                 79.8
# 3     3 1430700_a_at   Pla2g7                               65.1
# 4     4 1436611_at     Slc39a12                             62.2
# 5     5 1448139_at     Mlc1                                 58.6
# 6     6 1436231_at     AU067665                             57.9
# 7     7 1418937_at     Dio2                                 56.0
# 8     8 1428114_at     Slc14a1                              54.4
# 9     9 1424400_a_at   Aldh1l1                              54.0
# 10    10 1419559_at     Cyp4f14                              53.5
# ... with 2,608 more rows

## load in genesets

load_geneset <- function(file_path)
{
  geneset_tibble <- read_xls(path = file_path, skip = 1)
  # check input read in to match expectations
  cond1 <- colnames(geneset_tibble)[3] == 'Gene Name'
  cond2 <- grepl('Fold Enriched In', colnames(geneset_tibble)[4])
  # expect fold change value for gene to be >= 1.5
  cond3 <- all(geneset_tibble[,4] >= 1.5)
  genes <- geneset_tibble$`Gene Name`
  cond4 <- length(unique(genes)) == length(genes)
  if (! all(c(cond1, cond2, cond3, cond4)))
  {
    stop(paste('not all expected conditions met for input read from file',
               file_path))
  }
  return(genes)
}

cahoy.markersets <- list()
for (i in 1:length(names_use))
{
  names_use_i <- names_use[i]
  file_i <- xls_files[i]
  fpath <- file.path(cahoy.data.folder, file_i)
  cahoy.markersets[[names_use_i]] <- load_geneset(file_path = fpath)
}

### Map mouse genes to human genes
## define function to perform mapping
# human_mart <- useMart("ensembl", dataset = "hsapiens_gene_ensembl", 
#                       host = 'http://jan2019.archive.ensembl.org')
# mouse_mart <- useMart("ensembl", dataset = "mmusculus_gene_ensembl",
#                       host = 'http://jan2019.archive.ensembl.org')
human_mart <- useMart("ensembl", dataset = "hsapiens_gene_ensembl")
mouse_mart <- useMart("ensembl", dataset = "mmusculus_gene_ensembl")
mouse_2_human <- function(mouse_geneset, attr. = 'mgi_symbol', filt. = 'mgi_symbol',
                          attrL = 'hgnc_symbol')
{
  mouse_human_LDS <- getLDS(attributes = attr.,
                            filters = filt.,
                            values = mouse_geneset,
                            mart = mouse_mart,
                            attributesL = attrL,
                            martL = human_mart
  )
  # since you have cases of multiple mouse genes mapping
  # to single human genes, and multiple human genes mapping from
  # single mouse gene, take only unique human genes
  unique_human_genes <- unique(mouse_human_LDS$HGNC.symbol)
  return(unique_human_genes)
}
## perform conversion, save result
cahoy.markersets.hgnc <- list()
cahoy.markersets.hgnc <- lapply(cahoy.markersets, FUN = mouse_2_human)

## record geneset sizes for human and mouse genesets
gs_sizes_mouse <- as.numeric(lapply(cahoy.markersets, FUN = length))
names(gs_sizes_mouse) <- names(cahoy.markersets)
gs_sizes_human <- as.numeric(lapply(cahoy.markersets.hgnc, FUN = length))
names(gs_sizes_human) <- names(cahoy.markersets.hgnc)
gs_sizes_df <- data.frame(geneset_name = names(cahoy.markersets),
                          mouse_gs_size = gs_sizes_mouse,
                          human_gs_size = gs_sizes_human)
fname <- 'Cahoy_markerset_geneset_sizes_mouse_human.csv'
fpath <- file.path(output.dir, fname)
write.csv(gs_sizes_df, file = fpath)
rm(fname, fpath)

all.markers <- c(all.markers, cahoy.markersets.hgnc)
```

``` {r}
###############################################################################
### A1 A2 Astrocyte Markers, Zamanian et al. 2012, Genomic Analysis of
### Reactive Astrogliosis.

a1.a2.dir <- file.path(raw.data.folder, 'zamanian_2012_a1_a2')
a1.xls <- read_xlsx(file.path(a1.a2.dir, 'a1_astrocyte_LPS_inducted_genes.xlsx'))
a2.xls <- read_xlsx(file.path(a1.a2.dir, 'a2_astrocyte_MCAO_inducted_genes.xlsx'))
a1.a2.markersets <- list(a1.astro = a1.xls$`Gene symbol`,
                          a2.astro = a2.xls$`Gene symbol`)
# head(a1.a2.markersets)
# $a1.astro
# [1] "Lcn2"      "Steap4"    "H2-T23"    "Timp1"     "Serping1"  "Serpina3n" "H2-D1"     "Ggta1"     "Iigp1"    
# [10] "Gbp2"      "Cp"        "Fbln5"     "Ugt1a"     "Fkbp5"     "Aspg"      "Psmb8"     "Gbp10"     "Srgn"     
# [19] "Cxcl10"    "Amigo2"    "Hspb1"     "Osmr"      "Gfap"      "Ier3"      "Ifitm3"    "S1pr3"     "Sulf2"    
# [28] "Hspb6"     "Endou"     "Slc39a14"  "Tapbp"     "Crispld2"  "Gsr"       "Tspan4"    "Cxcl1"     "Gbp3"     
# [37] "Irgm1"     "Olfm1"     "H2-T10"    "Chi3l1"    "Vim"       "C4b"       "B2m"       "A2m"       "Slc22a4"  
# [46] "Nek6"      "Sorbs1"    "Gm3579"    "Cd44"      "Igtp"     
# 
# $a2.astro
# [1] "Lcn2"          "Ifi202b"       "Clcf1"         "Steap4"        "Tgm1"          "S1pr3"         "Ptx3"         
# [8] "S100a10"       "Sphk1"         "Timp1"         "Cd109"         "Ptgs2"         "Emp1"          "Slc10a6"      
# [15] "Tm4sf1"        "B3gnt5"        "Hspb1"         "Cd14"          "Galntl2"       "Igfbp3"        "Klf5"         
# [22] "Cxcl10"        "Cd44"          "Hmox1"         "Lgals3"        "Osmr"          "Gap43"         "Gdf15"        
# [29] "Flnc"          "Sbno2"         "Anxa2"         "Tubb6"         "Srxn1"         "Tnfrsf12a"     "Asns"         
# [36] "Ctgf"          "Gcnt2"         "Cp"            "Hmga1"         "Gadd45b"       "Vgf"           "Anxa3"        
# [43] "Serpina3n"     "Aspg"          "Vim"           "Odz2"          "Akap12"        "C730049O14Rik" "Gfap"         
# [50] "Icam1"        
a1.a2.markersets <- lapply(a1.a2.markersets, FUN = mouse_2_human)

all.markers <- c(all.markers, a1.a2.markersets)

```

``` {r}
###############################################################################
### Markersets from Mizrak 2019
mizrak.folder <- file.path(raw.data.folder, 'mizrak_2019')

load_sheets <- function(path, skip = 0) {
  all.sheets <- list()
  read_xlsx(path = path)
  for (sheet.name in excel_sheets(path)) {
    all.sheets[[sheet.name]] <- read_xlsx(path = path, sheet = sheet.name, skip = skip)
  }
  return(all.sheets)
}
## Tables s1 and s2. binomial specificity test as per Shekhar et al. 2016. 
## presumably the log2 effect represents an expression
## ratio (log2(fraction of expressing cells in a/ fraction of expressing cells
## not in a)). The gene sets here correspond to 'subclusters' they found in the paper.
mizrak.s1 <- load_sheets(file.path(mizrak.folder, '1-s2.0-S2211124718319740-mmc2.xlsx'))
# note that all gene lists are filtered for log2 effect > 0, FDR < 0.05
all(as.logical(lapply(mizrak.s1, FUN = function(x) all(x$log2_effect > 0))))
all(as.logical(lapply(mizrak.s1, FUN = function(x) all(x$fdr < 0.05))))
names(mizrak.s1)
mizrak.s2 <- load_sheets(file.path(mizrak.folder, '1-s2.0-S2211124718319740-mmc3.xlsx'))
all(as.logical(lapply(mizrak.s2, FUN = function(x) all(x$log2_effect > 0))))
all(as.logical(lapply(mizrak.s2, FUN = function(x) all(x$fdr < 0.05))))
mizrak.s2 <- lapply(mizrak.s2, FUN = function(x) return(x[which((x$log2_effect > 0) & (x$fdr < 0.05)),]))
names(mizrak.s2)
## table s3. filter genes at FDR 0.05, and separate upreg and downreg genes for each comparison.
mizrak.s3 <- load_sheets(file.path(mizrak.folder, '1-s2.0-S2211124718319740-mmc4.xlsx'), skip = 1)
mizrak.s3
mizrak.s3 <- mizrak.s3[-grep('Gsea', names(mizrak.s3))]
mizrak.s3.up <- lapply(mizrak.s3, FUN = function(x) return(x[which((x[,3] > 0) & (x$Fdr < 0.05)),]))
names(mizrak.s3.up) <- c('lateral_astro_M_upreg', 'septal_astrocytes_M_upreg', 'M_astro_lateral_upreg',
                         'F_astro_septal_upreg')
mizrak.s3.down <- lapply(mizrak.s3, FUN = function(x) return(x[which((x[,3] < 0) & (x$Fdr < 0.05)),]))
names(mizrak.s3.down) <- c('lateral_astro_F_upreg', 'septal_astrocytes_F_upreg', 'M_astro_septal_upreg',
                         'F_astro_lateral_upreg')
mizrak.s3 <- c(mizrak.s3.up, mizrak.s3.down)

mizrak.genesets <- c(mizrak.s1, mizrak.s2, mizrak.s3)
names(mizrak.genesets) <- paste('mizrak', names(mizrak.genesets), sep = '_')
## extract just the genes
# retrieve ENSMUSG IDs and  remove version numbers from symbols
mizrak.genesets <- lapply(mizrak.genesets, FUN = function(x) {
  # extract column with ENSMUSG ids w/version numbers
  col.use <- which(apply(x, MARGIN = 2, FUN = function(y) {
    return(all(grepl('ENSMUSG[0-9]+[.][0-9]+', y)))
  }))
  if (length(col.use) == 1) {
    return(as.character(x[[col.use]]))
  } else {
    stop('zero or multiplle columns with ENSMUSG identifiers')
  }
  
} )
mizrak.genesets <- lapply(mizrak.genesets, FUN = function(x) sub('[.][0-9]*', '', x))
lapply(mizrak.genesets, head)
## map to human symbols
mizrak.genesets <- lapply(mizrak.genesets, FUN = function(x) {
  mouse_2_human(mouse_geneset = x,
                attr. = c('ensembl_gene_id'),
                filt. = 'ensembl_gene_id',
                attrL = 'hgnc_symbol')
})
# for (x in names(mizrak.genesets)) {
#   print(x)
#   output <- mouse_2_human(mouse_geneset = mizrak.genesets[[x]],
#                           attr. = c('ensembl_gene_id'),
#                           filt. = 'ensembl_gene_id',
#                           attrL = 'hgnc_symbol')
# }

all.markers <- c(all.markers, mizrak.genesets)
```

``` {r}
###############################################################################
# ### Owen's markersets for SU2C C1, C2, shown in Oct 2018 committee meeting
# ## These were defined based on clusters obtained from RNA-seq data on
# ## Dirks and Weiss Lab GSC lines

### Owen's markersets for SU2C C1, C2, produced by analysis in RNA_GSCs_clustering_DE.Rmd
owen_markers_dir <- file.path(top.dir, 'results/RNA/RNA_GSCs_clustering_DE')
tryCatch({
  RNA.GSC.c1 <- read.csv(file.path(owen_markers_dir,
                                      'spectral_k2_c1_upreg_FDR_0_05.csv'),
                            stringsAsFactors = F)
  RNA.GSC.c2 <- read.csv(file.path(owen_markers_dir,
                                      'spectral_k2_c2_upreg_FDR_0_05.csv'),
                            stringsAsFactors = F)
  owen.RNA.k2.genesets <- list(RNA.GSC.c1 = as.character(RNA.GSC.c1$x),
                            RNA.GSC.c2 = as.character(RNA.GSC.c2$x))
  
  all.markers <- c(all.markers, owen.RNA.k2.genesets)
}, 
error = function(e) {
stop(paste('failed to load RNA GSC clustering based markersets. possibly could not find .csv files',
              'containing differentially expressed genes in', 
              owen_markers_dir, '; did you run RNA_GSCs_clustering_DE.Rmd?', sep = '\n'))
})


```

``` {r}
### Stem upregulated genes in a comparison of GSC (cancer stem cell) lines and
### Bulk Tumor
tryCatch({
  RNA.tissue.line.results.dir <- file.path(top.dir, 'results/RNA', 'RNA_tissue_line_clustering')
  glioma.stem <- read.csv(file.path(RNA.tissue.line.results.dir, "line_upreg_FDR_0_05.csv"))
  head(glioma.stem)
  glioma.stem <- as.character(glioma.stem$x)
  glioma.stem <- list(glioma.stem.cell = glioma.stem)
  
  all.markers <- c(all.markers, glioma.stem)
},
error = function(x) {
  stop(paste('could not find .csv file for GSC markers in', RNA.tissue.line.results.dir,
             '; did you run RNA_tissue_line_clustering_DE.Rmd'))
})


```


``` {r}
## Reduced Gene Sets for Stem, C1, C2, obtained by feature selection
## (l1 regularization in classification problem for GSC c1 vs c2 or
##  tissue vs line)
## GSC c1 classifier genes are a subset of RNA GSC c1 markers,
## GSC c2 classifier genes are a subset of RNA GSC c2 markers,
## GSC line classifier genes are a subset of glioma stem cell markers.
## In each case, corresponding original markerset was used in logistic
## regression with CV + L1 regularization. features with non-zero coefficients
## were selected from each model to make the 'classifier genes' gene sets
## For more details, see feature_selection_logistic.Rmd

tryCatch({
  RNA.feature.selection.dir <- file.path(top.dir, 'results/RNA/feature_selection/')
  file.list <- paste0(c('RNA_GSC_c1_upreg_', 'RNA_GSC_c2_upreg_', 'RNA_GSC_line_upreg_'),
                      'classifier_genes.csv')
  
  for (fname in file.list) {
    genes.table <- read.csv(file.path(RNA.feature.selection.dir, fname), stringsAsFactors = F)
    gset.name <- sub('_upreg_classifier_genes.csv', '_classifier', fname)
    genes.use <- list()
    genes.use[[gset.name]] <- genes.table$x
  }
  all.markers <- c(all.markers, genes.use)
  
},
error = function(x) {
  stop(paste('could not find .csv file for GSC markers in', RNA.tissue.line.results.dir,
             '; did you run feature_selection_logistic.Rmd?'))
})
```

``` {r}
## Tirosh et al. 2015 Cell Cycle genesets
tryCatch({
  # '../../../data/raw/GeneSets/regev_lab_cell_cycle_tirosh_2015/regev_lab_cell_cycle_genes.txt'
  cc_genes <- readLines(con = "~/projects/su2c_v2/data/raw/GeneSets/regev_lab_cell_cycle_tirosh_2015/regev_lab_cell_cycle_genes.txt")
  s_genes <- cc_genes[1:43]
  g2m_genes <- cc_genes[44:97]
  tirosh_cc_genesets <- list(tirosh_s_phase = s_genes, tirosh_g2m = g2m_genes)
  
  all.markers <- c(all.markers, tirosh_cc_genesets)
}, error = function(e) {
  stop('failed to load in + include tirosh 2015 cell cycle genesets')
})


```


``` {r}
## Neftel et al. 2019 GBM markers
tryCatch({
  neftel_markers_file <- "~/projects/su2c_v2/data/raw/GeneSets/neftel_2019/table_s2_neftel_2019_glioblastoma.xlsx"
  neftel_xls_contents <- as.data.frame(read_xlsx(neftel_markers_file, skip = 4, col_name = TRUE))
  neftel_genesets <- lapply(neftel_xls_contents, FUN = function(x) (return(x)))
  names(neftel_genesets) <- paste0('Neftel_', names(neftel_genesets))

  all.markers <- c(all.markers, neftel_genesets)
}, error = function(e) {
  stop('failed to load Neftel et al. 2019 GBM markers')
})


```

``` {r}
## Roulois et al. 2015 Interferon Response Markers
tryCatch({
  roulois_markers_file <-  "~/projects/su2c_v2/data/raw/GeneSets/Roulois_2015/DavidGenesSignature.txt"
  roulois_genes <- read.csv(roulois_markers_file, header = FALSE, stringsAsFactors = FALSE)
  roulois_genesets <- list(Roulois_Interferon_Response = roulois_genes$V1)

  all.markers <- c(all.markers, roulois_genesets)
}, error = function(e) {
  stop('failed to add roulois 2015 genesets')
})

```

``` {r}
# Neftel 2019, used to distinguish tumour from normal cells


tryCatch({
  neftel_genesets <- list()
  neftel_genesets[['Neftel_Macrophage']] <- c('CD14', 'AIF1', 'FCER1G', 'FCGR3A', 'TYROBP', 'CSF1R')
  neftel_genesets[['Neftel_T_Cell']] <- c('CD2', 'CD3D', 'CD3E', 'CD3G')
  neftel_genesets[['Neftel_Oligodendrocyte']] <- c('MBP', 'TF', 'PLP1', 'MAG', 'MOG', 'CLDN11')
  all.markers <- c(all.markers, neftel_genesets)
}, error = function(e) {
  stop('failed to add roulois 2015 genesets')
})
```

```{r}
###############################################################################
### save results to file

# all genes
genesets.and.info <- filter_genesets(all.markers, 
                                     master_geneset = NULL,
                                     min_geneset_size = 1, 
                                     max_geneset_size = Inf)
saveRDS(genesets.and.info, file = file.path(output.dir, 'genesets_and_info.rds'))
# genes filtered by being present in SU2C dds object
genesets_and_info_RNA_filt <- filter_genesets(all.markers, 
                                             master_geneset = SU2C.all.RNA.genes,
                                             min_geneset_size = 1, 
                                             max_geneset_size = Inf)
saveRDS(genesets_and_info_RNA_filt, file = file.path(output.dir, 'genesets_and_info_RNA_filt.rds'))


```

__Session Info__
```{r}
sessionInfo()
```
